rmw_connext_shared_cpp/src/node_names.cpp

```diff
@@ -93,12 +110,28 @@ get_node_names(
     if (name.empty()) {
       // ignore discovered participants without a name
       node_names->data[i] = nullptr;
+      node_namespaces->data[i] = nullptr;
       continue;
     }
+
     node_names->data[i] = rcutils_strdup(name.c_str(), allocator);
     if (!node_names->data[i]) {
       RMW_SET_ERROR_MSG("could not allocate memory for node name")
       rcutils_ret = rcutils_string_array_fini(node_names);
+      rcutils_ret = rcutils_string_array_fini(node_namespaces);
+      if (rcutils_ret != RCUTILS_RET_OK) {
+        RCUTILS_LOG_ERROR_NAMED(
+          "rmw_connext_cpp",
+          "failed to cleanup during error handling: %s", rcutils_get_error_string_safe())
+      }
+      return RMW_RET_BAD_ALLOC;
+    }
+
+    node_namespaces->data[i] = rcutils_strdup(namespace_.c_str(), allocator);
+    if (!node_namespaces->data[i]) {
+      RMW_SET_ERROR_MSG("could not allocate memory for node namespace")
+      rcutils_ret = rcutils_string_array_fini(node_names);
+      rcutils_ret = rcutils_string_array_fini(node_namespaces);
```

Comment:
If both fail then you'll need to reset the error state between them, but then again, perhaps that's ok or even intentional.

Separately I'll say that rather than duplicating this cleanup code, I'd recommend using a label and a goto to have it happen in one place.